### PR TITLE
Amesos2: Upgrade Cholmod for GPU

### PR DESCRIPTION
**Pull request by MicheldeMessieres**
Adds GPU capability to Cholmod solver.
Refactor Cholmod solver to use Kokkos Views.
Fix ETI for Cholmod.
Fix Complex for Cholmod.
Epetra now working for Cholmod.
Add long long GO test for Cholmod to match Trilinos defaults.
Add new Kokkos Adapter tests for Cholmod.
Setup/clean up Cholmod float but Cholmod does not support it yet.

@trilinos/amesos2

## Testing

Amesos2_Cholmod_Solver_Test

### CODE REVIEW

**Comment by srajama1** [comment]
@ndellingwood @iyamazaki will take the first pass at this PR.

File: packages/amesos2/src/Amesos2_Cholmod_TypeMap.hpp [hunk 1]
```diff
@@ -68,92 +65,13 @@
...
-
-
 namespace Amesos2 {
 
   // forward declaration due to circular reference
   template <class, class> class Cholmod;
 
   template <>
-  struct TypeMap<Cholmod,float>
+  struct TypeMap<Cholmod,float> // Cholmod does not support float yet
```
Comment (ndellingwood) [inline packages/amesos2/src/Amesos2_Cholmod_TypeMap.hpp]: @MicheldeMessieres what happens if someone tries to call this solver with float enabled, will it throw an error with a warning, silently convert to double, or silently fail?
Similar question for complex_float cases.

**Comment by ndellingwood** [comment]
@MicheldeMessieres I don't think cholmod will be tested by PR testing (KLU2 is enabled as the default solver), can you post your configuration and how you tested? Thanks!

File: packages/amesos2/src/Amesos2_Cholmod_TypeMap.hpp (hunk 1 above)
Comment (MicheldeMessieres) [reply packages/amesos2/src/Amesos2_Cholmod_TypeMap.hpp]: This will give a run time error that float is not supported.

**Comment by MicheldeMessieres** [comment]
A config for testing on white:

cmake \
  -GNinja \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
  -DAmesos2_ENABLE_TESTS=ON -DTrilinos_ENABLE_Amesos2=ON \
  -DAmesos2_ENABLE_Cholmod=ON \
  -DTPL_ENABLE_Cholmod=ON \
  -DCholmod_LIBRARY_DIRS="/ascldap/users/medemes/SuiteSparse/lib" \
  -DCholmod_LIBRARY_NAMES="cholmod,camd,ccolamd,colamd,suitesparseconfig,amd" \
  -DCholmod_INCLUDE_DIRS="/ascldap/users/medemes/SuiteSparse/include" \
  -DTPL_ENABLE_METIS=ON \
  -DTPL_ENABLE_MPI=ON \
  -DTrilinos_ENABLE_Epetra=ON \
  -DTrilinos_ENABLE_EpetraExt=ON \
  -DAmesos2_ENABLE_Experimental=ON \
  ../trilinospath

This would use the white installed Metis. SuiteSparse can install its own metis so alternative is to set the Metis paths for that. If SuiteSparse has metis.h and you are using white's metis, then there can be conflicts. Cholmod solver won't exhibit a problem but Tacho solver (or the upcoming cuSolver solver) will crash on the metis sort, for example.

For installing SuiteSparse on white, I set ENV as follows:
export BLAS=“-L/home/projects/ppc64le-pwr8/netlib/3.8.0/gcc/7.2.0/lib -lblas -lgfortran -lgomp”
export LAPACK=“-L/home/projects/ppc64le-pwr8/netlib/3.8.0/gcc/7.2.0/lib -llapack -lgfortran -lgomp”
export MY_METIS_LIB=“/home/projects/ppc64le-pwr8/metis/5.0.1/gcc/7.2.0/lib/libmetis.a”
export MY_METIS_INC=“/home/projects/ppc64le-pwr8/metis/5.0.1/gcc/7.2.0/include”

Then make config can check the status before make.

Note trying to set things like MY_METIS_LIB in the make line directly, instead of by export first, as indicated in the docs, didn't seem to work.

**Pull request merged**